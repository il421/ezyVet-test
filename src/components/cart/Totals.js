import React, { Component } from 'react';
import { formatCurrency } from '../../libs/currency.js';
import '../../styles/cart/totals.css';

export class Totals extends Component {

  getTotalAmount = (cart) => {
    let total = 0;

    for (let i = 0; i < cart.length; i++) {
      const item = cart[i];
      total += item.price * item.qty;

    }
    return formatCurrency(total);
  }

  render() {
    return (
      <div className="totals">
        <div>Totals: { this.getTotalAmount(this.props.cart) }</div>
      </div>
    );
  }
}

export default Totals;
