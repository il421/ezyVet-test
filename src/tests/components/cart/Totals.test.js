import React from 'react';
import { shallow } from 'enzyme';
import Totals from '../../../components/cart/Totals.js';
import { cart } from '../../fixtures/data';

let wrapper;

beforeEach(() => {
  wrapper = shallow(
    <Totals
      cart={ cart }
    />
  );
});

test('should render Totals without crashing with correct totals value', () => {
  const totals = 'Totals: $316.25';
  expect(wrapper).toMatchSnapshot();
  expect(wrapper.find('.totals div').text()).toBe(totals);
});
