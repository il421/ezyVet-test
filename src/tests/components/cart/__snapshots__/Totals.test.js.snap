// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Totals without crashing with correct totals value 1`] = `
<div
  className="totals"
>
  <div>
    Totals: 
    $316.25
  </div>
</div>
`;
